.github/workflows/merge.yml

```
@@ -1,16 +1,14 @@
-name: Backend CI/CD
+name: Backend develop CI/CD
 
 on:
   workflow_dispatch:
   
-  pull_request:
+  push:
```

P2. 혹시 push로 설정해도 merge될 때만 반영이 되는지 궁금합니다!
만약 안된다면, 아래 처럼 설정하는 것을 제안합니다~

```suggestion
  pull_request:
    branches: [ develop ]
    types: [closed] // 브랜치를 닫을 때에만 실행
    paths: backend/**
    
    
permissions:
  contents: read
  
jobs:
  build-and-upload:

    runs-on: ubuntu-22.04
    if: github.event.pull_request.merged // 닫은 브랜치가 머지되었을 때만 실행(단순히 브랜치 닫기만 한 경우 실행 방지)
```

주석 단 부분만 다른 부분입니다.

print job으로 [테스트](https://github.com/yoondgu/github-action-practice/actions/runs/5668217175/job/15358485411)해본 결과 merge가 아닐 때는 job이 skip되고, merge일 때는 실행되는 것 확인했습니다!

---

아 직접 push하지 않고 merge를 통해 새 커밋이 추가된 경우에도 push 이벤트가 사용 가능하네요!

PR을 거치지 않고 develop 브랜치에 직접 push한 경우에도 자동 배포를 허용할 것인지에 따라 두 방식 중 하나로 결정하면 될 것 같습니다!

---

@yoondgu 존경합니다 도이..